doc/source/notebooks/manual/basics/classification.ipynb

```diff
@@ -4,11 +4,299 @@
+    "plt.plot(x_grid, f_grid, 'C0', linewidth=1)\n",
+    "plt.plot(x_grid, f_grid[:, 1], 'C0', linewidth=2);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "__2. Squash them to $[0, 1]$:__ the samples of the GP are mapped to $[0, 1]$ using the logistic inverse link function: $g(x) = \\frac{\\exp(f(x))}{1 + \\exp(f(x))}$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "p_grid = np.exp(f_grid) / (1 + np.exp(f_grid))\n",
+    "plt.plot(x_grid, p_grid, 'C1', linewidth=1)\n",
+    "plt.plot(x_grid, p_grid[:, 1], 'C1', linewidth=2);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "__3. Sample from a Bernouilli:__ for each observation point $X_i$, the class label $Y_i \\in \\{0, 1\\}$ is generated by sampling from a Bernouilli distribution $Y_i \\sim \\mathcal{B}(g(X_i))$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Select some input locations\n",
+    "ind = np.random.randint(0, 200, (30,))\n",
+    "X_gen = x_grid[ind]\n",
+    "\n",
+    "# evaluate probability and get bernoulli draws\n",
+    "p = p_grid[ind, 1:2]\n",
+    "Y_gen = np.random.binomial(1, p)\n",
+    "\n",
+    "#plot\n",
+    "plt.plot(x_grid, p_grid[:, 1], 'C1', linewidth=2)\n",
+    "plt.plot(X_gen, p, 'C1o', ms=6)\n",
+    "plt.plot(X_gen, Y_gen, 'C3x', ms=8, mew=2);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Implementation with GPflow\n",
+    "\n",
+    "For the model described above, the posterior $f(x)|Y$ (say $p$) is not gaussian anymore and does not have a closed form expression. A common approach is then to look for the best approximation of this posterior by a tractable distribution (say $q$) such as a Gaussian distribution. In variational inference, the quality of an approximation is measured by the KL divergence between $KL[q || p]$. For more details on this model, see Nickisch et al (2008).\n",
+    "\n",
+    "The inference problem is thus turned into an optimisation problem: finding the best parametres for $q$. In our case, we introduce $U \\sim \\mathcal{N}(q_\\mu, q_\\Sigma)$ we choose $q$ to have the same distribution as $f | f(X) = U$. The parameters $q_\\mu$ and $q_\\Sigma$ can be seen as parametres of $q$, which can be optimised in order to minimise  $KL[q || p]$. \n",
+    "\n",
+    "This variational inference model is called `VGP` in GPflow:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = gpflow.models.VGP(X, Y,\n",
+    "                      likelihood=gpflow.likelihoods.Bernoulli(),\n",
+    "                      kern=gpflow.kernels.Matern52(1))\n",
+    "\n",
+    "o = gpflow.train.ScipyOptimizer()\n",
+    "o.minimize(m);"
```

Comment:
can you add a `maxiter` arg to the optimiser please
```
from gpflow.test_util import notebook_niter
o.minimize(m, maxiter=notebook_niter(10e3))
```

Reply:
I'd like the code to be as neat as possible so I would be in favour of keeping it the way it is. If it happens to be too slow on the servers, I am happy to reduce the number of observation points, and to stop this optim before convergence.